interval/search_test.go

```diff
@@ -587,6 +590,60 @@ func TestSearchTree_Select(t *testing.T) {
 	}
 }
 
+func TestSearchTree_EncodingDecoding(t *testing.T) {
+	var b bytes.Buffer
+	w := bufio.NewWriter(&b)
+	r := bufio.NewReader(&b)
+
+	st1 := NewMultiValueSearchTree[string](func(x, y int) int { return x - y })
+	defer mustBeValidTree(t, st1.root)
+
+	st2 := NewMultiValueSearchTree[string](func(x, y int) int { return x - y })
+	defer mustBeValidTree(t, st2.root)
```

Comment:
What do you think of adding a test case for trees with options too?
```suggestion
	st1 := NewMultiValueSearchTree[string](func(x, y int) int { return x - y }, TreeWithIntervalPoint())
	defer mustBeValidTree(t, st1.root)

	st2 := NewMultiValueSearchTreeWithOptions[string](func(x, y int) int { return x - y }, TreeWithIntervalPoint())
	defer mustBeValidTree(t, st2.root)
```

And then compare if the option is equally set on both trees:
```go
	// Config should be equal
	if !reflect.DeepEqual(st2.config.allowIntervalPoint, st1.config.allowIntervalPoint) {
		t.Fatal("Trees are not equal")
	}
```

Reply:
Done: [Add more tests](https://github.com/rdleal/intervalst/pull/9/commits/28e2d47e25ff6bb7b8d847dbe5c0bbe085dbc91d)

I've also added tests for empty trees as well, since there was a bug when encoding an empty tree (fixed in [Allow empty trees to be encoded](https://github.com/rdleal/intervalst/pull/9/commits/f6ecf51e1a33d243fe44fcc6c414af808c1868d3)).